src/btree/bt_read.c

```diff
@@ -152,6 +152,78 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
     return (ret);
 }
 
+/*
+ * __wt_modify_vector_init --
+ *     Initialise a modify vector.
+ */
+void
+__wt_modify_vector_init(WT_MODIFY_VECTOR *modifies, WT_SESSION_IMPL *session)
+{
+    WT_CLEAR(*modifies);
+    modifies->session = session;
+    modifies->listp = modifies->list;
+}
+
+/*
+ * __wt_modify_vector_push --
+ *     Push a modify update pointer to a modify vector. If we exceed the allowed stack space in the
+ *     vector, we'll be doing malloc here.
+ */
+int
+__wt_modify_vector_push(WT_MODIFY_VECTOR *modifies, WT_UPDATE *upd)
+{
+    WT_DECL_RET;
+    bool migrate_from_stack;
+
+    migrate_from_stack = false;
+
+    if (modifies->size >= WT_MODIFY_VECTOR_STACK_SIZE) {
+        if (modifies->capacity == 0 && modifies->size == WT_MODIFY_VECTOR_STACK_SIZE) {
+            migrate_from_stack = true;
+            modifies->listp = NULL;
+        }
+        WT_ERR(__wt_realloc_def(
+          modifies->session, &modifies->capacity, modifies->size + 1, &modifies->listp));
+        if (migrate_from_stack)
+            memcpy(modifies->listp, modifies->list, sizeof(modifies->list));
+    }
+    modifies->listp[modifies->size++] = upd;
+    return (0);
```

Comment:
I did it like this because you might have a case where you might have a `push`, `pop`, `push` pattern where you cross the `WT_MODIFY_VECTOR_STACK_SIZE` multiple times.

Reply:
I agree with your argument.